### PR TITLE
CNDIT-1501: Modify Investigation postprocessing for datamart topic

### PR DESCRIPTION
**Pull request by sveselev**
## Description

This PR includes modification of `sp_nrt_investigation_postprocessing` to return Datamart specific columns and Java post-processing service to accept it.

Post-processing service:
- **InvestigationResult.java** (renamed from `InvestigationStoredProc`): Entity that contains columns to produce Datamart specific topic
- **InvestigationRepository.java**: modified to adopt `sp_nrt_investigation_postprocessing` result set
- **PostProcessingService.java**: modified to call InvestigationRepository method and accept the result

Liquibase:
- **023-sp_nrt_investigation_postprocessing-001.sql**: SELECT for Datamart topic columns added to sp

## JIRA
[CNDIT-1501](https://cdc-nbs.atlassian.net/browse/CNDIT-1501)

### CODE REVIEW

File: data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java [hunk 1]
```diff
@@ -37,7 +39,12 @@ public class PostProcessingService {
     private final PageBuilderRepository pageBuilderRepository;
 
     static final String PAYLOAD = "payload";
-    static final String SP_EXECUTION_COMPLETED = "Stored proc execution completed.";
+    static final String INVESTIGATION = "investigation";
```
Comment (alog-enquizit) [inline data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java:42]: Do you think we can make it as an `enum`

**Review by alog-enquizit — APPROVED** [review]
lgtm 👍

File: data-reporting-service/src/main/resources/db/rdb/023-sp_nrt_investigation_postprocessing-001.sql [hunk 2]
```diff
@@ -1,464 +1,501 @@
...
+           COALESCE(inv.INVESTIGATION_KEY, 1) AS investigation_key,
+           COALESCE(pat.PATIENT_KEY, 1) AS patient_key,
+           nrt.cd AS condition_cd,
+           dtm.Datamart AS datamart,
+           dtm.Stored_Procedure AS stored_procedure
+    FROM dbo.nrt_investigation nrt
+             LEFT JOIN INVESTIGATION inv ON inv.CASE_UID = nrt.public_health_case_uid
+             LEFT JOIN D_PATIENT pat ON pat.PATIENT_UID = nrt.patient_id
+             LEFT JOIN dbo.nrt_datamart_metadata dtm ON dtm.condition_cd = nrt.cd
+    WHERE nrt.public_health_case_uid in
```
Comment (upasanapattnaik-eq) [inline data-reporting-service/src/main/resources/db/rdb/023-sp_nrt_investigation_postprocessing-001.sql:472]: LGTM!

**Review by upasanapattnaik-eq — APPROVED** [review]
LGTM! Thank you for implementing :)

File: data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java (hunk 1 above)
Comment (sveselev) [reply data-reporting-service/post-processing-service/src/main/java/gov/cdc/etldatapipeline/postprocessingservice/service/PostProcessingService.java:42]: @alog-enquizit yes, I even tried an implementation, but there is a problem with passing the repositories, in a construction like:

```
enum TopicProcessor {
    ORGANIZATION(ORGANIZATION, "sp_nrt_organization_postprocessing") {
        @Override
        void process(String keyTopic, List<Long> ids) {
            processTopic(keyTopic, ids, organizationRepository::executeStoredProcForOrganizationIds, this.topic, this.storedProc);
        }
    },
    PROVIDER(PROVIDER, "sp_nrt_provider_postprocessing") {
        @Override
        void process(String keyTopic, List<Long> ids) {
            processTopic(keyTopic, ids, providerRepository::executeStoredProcForProviderIds, this.topic, this.storedProc);
        }
    },
```
but repos and their methods are not static... I will think a bit more in the next implementation.

**Pull request merged**